### PR TITLE
feat(compass-indexes): add vector search index creation to create search indexes modal COMPASS-7302

### PR DESCRIPTION
**Pull request by Anemy**
COMPASS-7302

[Figma Designs](https://www.figma.com/file/vlBvSsZ0ZGEFNleOWoi8hd/EXPO-3047-Compass-Vector-Search?type=design&node-id=1%3A35&mode=dev)

Vector Search Indexes are only available on MongoDB versions 6.0.11+, 7.0.2+ in Atlas. For testing I use the `analytics-node-test` deployment in our Compass team project.

Screenshot 2024-02-04 at 10 54 04 AM

Screenshot 2024-02-04 at 11 07 46 AM

### CODE REVIEW

File: packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx
Comment (Anemy) [inline packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx:376]: I'll double check with product folks on these links before merging.

File: packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx
Comment (addaleax) [inline packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx:51]: nit: I know this is covered through tests, but it might be a bit nicer to handle this through a property on the `template` itself rather than to rely on the names of specific templates?

File: packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx
Comment (addaleax) [inline packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx:322]: nit: I like the idea of keeping the `searchIndexTypes` list instead of just hardcoding two options as radio boxes here, but then the logical next step for me would be to also decouple the checks for the absense/presence of individual types from the React render tree here, e.g. filter `searchIndexTypes` at the top of the component and then just do something along the lines of 
```suggestion
              {filteredSearchIndexTypes.length > 1 && (
```

here

**Review by addaleax — APPROVED** [review]

File: packages/compass-indexes/src/components/search-indexes-modals/update-search-index-modal.tsx
Comment (gribnoysup) [inline packages/compass-indexes/src/components/search-indexes-modals/update-search-index-modal.tsx:69]: Should we do the `find` only once instead of two times, here and on the line above?

**Review by gribnoysup — APPROVED** [review]

File: packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx
Comment (Anemy) [reply packages/compass-indexes/src/components/search-index-template-dropdown/index.tsx:51]: I agree, I wasn't quite sure what I wanted to name that property though, `isReplacedByVectorSearch`?

File: packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx
Comment (Anemy) [reply packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx:322]: If it's alright with you I'd prefer to hold off on that refactor until theres more complexity to handle here, like more atlas search index types.

File: packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx
Comment (Anemy) [reply packages/compass-indexes/src/components/search-indexes-modals/base-search-index-modal.tsx:376]: Asked on slack: https://mongodb.slack.com/archives/C05C8MVJJQ7/p1707156642543119 as these links are probably fine for now I think I'll merge this pr and then update in a follow up if there are changes.

**Pull request merged**
